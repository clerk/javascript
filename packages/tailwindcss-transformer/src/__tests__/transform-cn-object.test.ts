import { transform } from '../index';

test('className with cn/cx/clsx', () => {
  const result = transform(
    `const ButtonCn: React.FC<ButtonProps> = ({ children, intent, className }) => {
      return <button className={cn({ danger: 'bg-red-500', 'success': 'bg-green-500' }[intent])}>{children}</button>
    }
    
    const ButtonCx: React.FC<ButtonProps> = ({ children, intent, className }) => {
      return <button className={cx({ danger: 'bg-red-500', 'success': 'bg-green-500' }[intent])}>{children}</button>
    }
    
    const ButtonClsx: React.FC<ButtonProps> = ({ children, intent, className }) => {
      return <button className={clsx({ danger: 'bg-red-500', 'success': 'bg-green-500' }[intent])}>{children}</button>
    }`,
    {
      styleCache: new Map(),
    },
  );
  expect(result).toMatchSnapshot();
});
