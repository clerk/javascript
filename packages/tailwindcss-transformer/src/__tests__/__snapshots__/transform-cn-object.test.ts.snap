// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`className with cn 1`] = `
"const Button: React.FC<ButtonProps> = ({ children, intent, className }) => {
    return <button className={cn({ danger: "cl-5a7355a7", 'success': "cl-c14c13cf" }[intent])}>{children}</button>;
  }"
`;
